sstables/sstables.cc

```diff
@@ -789,9 +789,9 @@ void write(sstable_version_types v, file_writer& out, const utils::streaming_his
     write(v, out, max_bin_size, a);
 }
 
-future<> parse(sstable_version_types v, random_access_reader& in, commitlog_interval& ci) {
-    return parse(v, in, ci.start).then([&ci, v, &in] {
-        return parse(v, in, ci.end);
+future<> parse(schema_ptr s, sstable_version_types v, random_access_reader& in, commitlog_interval& ci) {
+    return parse(s, v, in, ci.start).then([&ci, v, s, &in] {
+        return parse(s, v, in, ci.end);
```

Comment:
std::move(s) here (unfortunately I don't know what we can do about the two copies in the line above).